src/main/java/it/reply/orchestrator/dto/deployment/DeploymentMessage.java

```diff
@@ -74,6 +74,25 @@ public void setTimeoutInMins(Integer timeoutMins) {
   private boolean pollComplete;
   private boolean skipPollInterval;
 
+  @NonNull
+  @NotNull
+  private String providerTimeout = MAX_TIMEOUT.toString();
+
+  /**
+   * Sets the Deployment per provider timeout.
+   *
+   * @param timeoutMins
+   *          the timeout in Minutes
+   */
+  public void setProviderTimeoutInMins(Integer timeoutMins) {
+    this.providerTimeout = Optional
+        .ofNullable(timeoutMins)
+        .map(value -> Instant.now().plus(Duration.ofMinutes(value)))
```

Comment:
Why are you calculating this value one time and at request time?

Reply:
the value is assigned in the request so that is the time to calculate it

Reply:
you are calculating a fixed instant in time as `requestTime` + `providerTimeout`. That instant will be the same for every provider, so you don't actually have a timeout for a specific provider

Reply:
absolutely correct!